@relation directive shell

AmanJaveriKadri opened the pull request:
Issue #, if available:

Description of changes:
Implemented some of the initial setup for the @relation directive with a bunch of input checks. Confirms that the relation has been setup (mostly) correctly. There are also unit tests for the input checks added. (The directive is incomplete.)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

nau88xj committed:
Fix for invalid JSON in pipeline resolver

The JSON generated for the pipeline resolver is invalid and results in an error when sending any requests.  The two fields removed appear to be DynamoDB specific, so I removed them since we are resolving to a Lambda function.

AmanJaveriKadri committed:
feat(graphql-relation-transformer): added shell for relation directive

the relation directive was added with some basic input checks as well as tests for each input check.

mikeparisstuff commented on packages/graphql-relation-transformer/package.json:
```diff
@@ -0,0 +1,40 @@
+    "version": "1.0.0",
+    "description": "Implements the @relation directive.",
+    "main": "lib/index.js",
+    "author": "Aman Javeri Kadri",
```
Should be: `"author": "Amazon Web Services",`

mikeparisstuff commented on packages/graphql-relation-transformer/package.json:
```diff
@@ -0,0 +1,40 @@
+    "dependencies": {
+      "graphql": "^0.13.2",
+      "graphql-transformer-common": "3.8.2",
+      "graphql-transformer-core": "3.7.4"
```
Does this depend on core? You shouldn't need this.

mikeparisstuff commented on packages/graphql-relation-transformer/src/ModelRelationTransformer.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+}
+
+export default class RelationTransformer extends Transformer {
```
Please add comments explaining basic usage and purpose.

mikeparisstuff commented on packages/graphql-relation-transformer/src/ModelRelationTransformer.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+    const baseType = getBaseType(field.type);
+    const isAList = isListType(field.type)
+    // The only valid key fields are single String and ID fields.
```
Not technically true. We try to allow anything DynamoDB can index. See for example of how @key does this https://github.com/aws-amplify/amplify-cli/blob/bbe55bf6cdc626270ba738fdd5f2fbc33277525b/packages/graphql-key-transformer/src/KeyTransformer.ts#L377

mikeparisstuff reviewed (commented):
Early comments

mikeparisstuff committed:
Removing api.valid.migration.ts and moving tests into other files so they run serially. (#1929)

kaustavghosh06 committed:
feat: adding amplify cli predictions category (#1936)

kaustavghosh06 committed:
chore(release): Publish [ci skip] (#1940)

- amplify-category-analytics@1.6.0
 - amplify-category-api@1.8.0
 - amplify-category-auth@1.8.0
 - amplify-category-function@1.8.0
 - amplify-category-hosting@1.3.0
 - amplify-category-interactions@1.7.0
 - amplify-category-notifications@1.5.0
 - amplify-category-predictions@1.1.0
 - amplify-category-storage@1.9.0
 - amplify-category-xr@1.7.0
 - @aws-amplify/cli@1.9.0
 - amplify-codegen@1.6.0
 - amplify-e2e-tests@1.7.0
 - amplify-frontend-android@1.9.0
 - amplify-frontend-ios@1.9.0
 - amplify-frontend-javascript@1.7.0
 - amplify-graphql-docs-generator@1.5.0
 - amplify-graphql-types-generator@1.1.0
 - amplify-provider-awscloudformation@1.14.0
 - graphql-auth-transformer@3.8.0
 - graphql-connection-transformer@3.6.0
 - graphql-dynamodb-transformer@3.9.0
 - graphql-elasticsearch-transformer@3.7.0
 - graphql-function-transformer@1.2.0
 - graphql-http-transformer@3.5.0
 - graphql-key-transformer@1.2.0
 - graphql-mapping-template@3.2.0
 - graphql-relational-schema-transformer@1.1.0
 - graphql-transformer-common@3.9.0
 - graphql-transformer-core@3.8.0
 - graphql-transformers-e2e-tests@3.7.0
 - graphql-versioned-transformer@3.5.0

rohitchandalurivt committed:
Showing message for non admin task in identify entities category

rohitchandalurivt committed:
Renaming varaible and changing the message of display

Commit:
Merge pull request #1942 from rohitchandalurivt/master

Showing message for non admin task in identify entities category

AmanJaveriKadri committed:
feat(graphql-relation-transformer): added get item and style changes

wrote function to configure the resolver needed to run a getItem for when the relation is using the
default table and providing the full primary key. Added comments explaining directive purpose and
use, refactored the validate field function and updated the package.json (all based on pull request
comments from Paris). Added comments to functions that were left uncommented and rearranged some of
the code for stylistic reasons.

kaustavghosh06 committed:
fix: fixing force push on init of new env (#1949)

fix #1945

ChristopheBougere committed:
style(amplify-codegen): eslint-disable comment style (#1912)

eslint-disable tag must be inside an /* */ comment block, not //

fossamagna committed:
fix(amplify-category-storage): remove unnecessary comma (#1755)

kaustavghosh06 committed:
chore(release): Publish [ci skip] (#1958)

- amplify-category-analytics@1.7.0
 - amplify-category-api@1.9.0
 - amplify-category-auth@1.9.0
 - amplify-category-function@1.9.0
 - amplify-category-hosting@1.4.0
 - amplify-category-interactions@1.8.0
 - amplify-category-notifications@1.6.0
 - amplify-category-predictions@1.2.0
 - amplify-category-storage@1.10.0
 - amplify-category-xr@1.8.0
 - @aws-amplify/cli@1.10.0
 - amplify-codegen@1.7.0
 - amplify-e2e-tests@1.8.0
 - amplify-frontend-android@1.10.0
 - amplify-frontend-ios@1.10.0
 - amplify-frontend-javascript@1.8.0
 - amplify-graphql-docs-generator@1.6.0
 - amplify-graphql-types-generator@1.2.0
 - amplify-provider-awscloudformation@1.15.0
 - graphql-auth-transformer@3.9.0
 - graphql-connection-transformer@3.7.0
 - graphql-dynamodb-transformer@3.10.0
 - graphql-elasticsearch-transformer@3.8.0
 - graphql-function-transformer@1.3.0
 - graphql-http-transformer@3.6.0
 - graphql-key-transformer@1.3.0
 - graphql-mapping-template@3.3.0
 - graphql-relational-schema-transformer@1.2.0
 - graphql-transformer-common@3.10.0
 - graphql-transformer-core@3.9.0
 - graphql-transformers-e2e-tests@3.8.0
 - graphql-versioned-transformer@3.6.0

rohitchandalurivt committed:
fix(amplify-category-predictions): changing the predictions lambda trigger time out to 15 mins (#1956)

Commit:
refactor(amplify-category-predictions): changed question wording (#1957)

changed wording when when selecting a predictions category

dkwl93 committed:
fix: fix #1205 (#1961)

palpatim committed:
Fix space in prompt (#1964)

haverchuck committed:
fix(amplify-category-auth): consolidates parameters below limit (#1948)

AmanJaveriKadri committed:
feat(graphql-relation-transformer): added query functionality

added code to configure resolvers to run dynamodb queries that return a list of objects. This
enables users to use relation to create relations to many objects rather than just one as with
getItem. Also added a test to check that the resolver has the right components and that that the
field is updated correctly.

kaustavghosh06 committed:
fix(amplify-e2e-tests): fixing predictions e2e tests (#1969)

yuth committed:
feat: mock support for API, function and storage (#1893)

Storage  and API category can run locally on developer machine without pushing these resources to cloud. Lambda functions can be executed locally using mock function command.

Added a new command `amplify mock` which runs provisioned AppSync GraphQL API, S3 storage and Lambda functions in developers computer without pushing them to the cloud.

yuth committed:
fix: local mock fix (#1982)

* chore(amplify-velocity-template): remove autogenrated file from git

Jison autogenerates parse.js. removing it from git

* ci: reduce concurrency of ci tests

* fix(amplify-frontend-javascript): prevent presevering test exports

* fix(amplify-appsync-simulator): show the complete VTL compile error

* feat(amplify-codegen): generate introspection schema for android proj

Android codegen uses a gradle plugin which needs introspection schema to generate the Java classes.
When mocking API we can't pull the introspection from AppSync service.

Updated codegen to generate introsepction schema from local SDL schema when the project is an
Android project

* refactor: update prompt messages

Commit:
fix(graphql-key-transformer): added sort direction

added sort direction argument in query list field

#1676

Commit:
fix(graphql-dynamodb-transformer): added scan index forward

add scan index forward for query expressions and added direction based on ModelSortDirection

#1676

Commit:
test(graphql-transformers-e2e-tests): added sort direction test

added sort direction e2e test

#1676

Commit:
refactor(amplify-velocity-template): updated vtl

updated vtl on setup

yuth committed:
build: move graphqljs dep to root package (#1985)

yuth committed:
Local mock use ip (#1989)

* fix(amplify-appsync-simulator): use local ip address for endpoint

Android emulator treats localhost as its own ip and doesn't forward the request to the host machine.
Updating appsync simulator to use the host ip address instead of localhost in endpoint

* docs(amplify-appsync-simulator): change subscription log msg

Commit:
updated branch from master

Commit:
updated key transformer test

removed unused import

kaustavghosh06 committed:
feat: adding post-install script (#1991)

* fixes based on PR comments

* fix: fix async push

* feat: adding post-install script

mikeparisstuff commented:
I don't see any updates related to @auth. Do you plan on addressing those in this PR? We will not be able to release the feature without auth support.

AmanJaveriKadri commented:
You're right, I haven't put that in yet. I'll add that in by tomorrow.


From: Michael
Sent: Wednesday, August 7, 1:47 PM
Subject: Re: [mikeparisstuff/amplify-cli] @relation directive shell  (#7)
To: mikeparisstuff/amplify-cli
Cc: Javeri Kadri, Aman, Author


I don't see any updates related to @auth. Do you plan on addressing those in this PR? We will not be able to release the feature without auth support.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mikeparisstuff/amplify-cli/pull/7?email_source=notifications&email_token=AGKWGWOHQPOARWQGIAGGVB3QDMYFBA5CNFSM4IFKYV42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3ZVD2Y#issuecomment-519262699>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AGKWGWPD24VGI3YEUG3ZTA3QDMYFBANCNFSM4IFKYV4Q>.

AmanJaveriKadri committed:
feat(graphql-relation-transformer): added queries by sort key

The queries did not yet support querying by partition key and sort key. Moreover adding a sort key
as an argument was not configured. This functionality was added.

kaustavghosh06 committed:
chore(release): Publish [ci skip] (#1993)

- amplify-appsync-simulator@0.2.0
 - amplify-category-analytics@1.8.0
 - amplify-category-api@1.10.0
 - amplify-category-auth@1.10.0
 - amplify-category-function@1.10.0
 - amplify-category-hosting@1.5.0
 - amplify-category-interactions@1.9.0
 - amplify-category-notifications@1.7.0
 - amplify-category-predictions@1.3.0
 - amplify-category-storage@1.11.0
 - amplify-category-xr@1.9.0
 - @aws-amplify/cli@1.11.0
 - amplify-codegen@1.8.0
 - amplify-dynamodb-simulator@0.2.0
 - amplify-e2e-tests@1.9.0
 - amplify-frontend-android@1.11.0
 - amplify-frontend-ios@1.11.0
 - amplify-frontend-javascript@1.9.0
 - amplify-graphiql-explorer@0.2.0
 - amplify-graphql-docs-generator@1.7.0
 - amplify-graphql-types-generator@1.3.0
 - amplify-provider-awscloudformation@1.16.0
 - amplify-storage-simulator@0.2.0
 - amplify-util-mock@0.2.0
 - amplify-velocity-template@0.1.0
 - graphql-auth-transformer@3.10.0
 - graphql-connection-transformer@3.8.0
 - graphql-dynamodb-transformer@3.11.0
 - graphql-elasticsearch-transformer@3.9.0
 - graphql-function-transformer@1.4.0
 - graphql-http-transformer@3.7.0
 - graphql-key-transformer@1.4.0
 - graphql-mapping-template@3.4.0
 - graphql-relational-schema-transformer@1.3.0
 - graphql-transformer-common@3.11.0
 - graphql-transformer-core@3.10.0
 - graphql-transformers-e2e-tests@3.9.0
 - graphql-versioned-transformer@3.7.0

nikhname committed:
fix(graphql-connection-transformer): fix self connection bug (#1944)

* fix(graphql-connection-transformer): fix self connection bug

fix #1799

* test(graphql-connection-transformer): test for self connections

* style: white-space and formatting

visusnet committed:
feat(amplify-provider-awscloudformation): hooks (#1951)

Every function resource can have a custom build script in the main package.json with the script name amplify:<resource name> which is executed after npm install.

kaustavghosh06 committed:
feat: narrow-down idp roles scope (#1974)

* fixes based on PR comments

* fix: fix async push

* feat: narrow-down idp roles scope

* narrow down IAM update plicies for the lambda execution role

yuth committed:
fix(amplify-util-mock): handle unsupported data source gracefully (#1999)

API Mock server did not handle unsupported data source such as elastic search well. Updated the code
to handle it gracefully and use NONE as the data source when not supported

fix #1997

yuth committed:
fix(amplify-graphql-types-generator): generate valid swift code

Codegen automatically adds localUri and mimeTypes for S3Object.
Prevent adding these fields if they are already declared in S3ObjectInput

fix #1903

warrenmcquinn committed:
fix(amplify-provider-awscloudformation): apigw unauth access (#1906)

AmanJaveriKadri committed:
test(graphql-relation-transformer): basic e2e tests for relation

Added an E2E test for a getItem connection (relation to a single object) and a test for a relation
that uses a query (a relation to multiple objects).

Commit:
refactor(graphql-dynamodb-transformer): typo in req resolver

removed double $ in the list query resolver request

#1676 #1990 #1629

Commit:
fix(graphql-key-transformer): key req resolver edit

added sort direction check in the key transformer request resolver

#1676 #1990 #1629

Commit:
test(graphql-transformers-e2e-tests): added gsi e2e test

added sort direction e2e test on @key with names and queryfields

#1676 #1990 #1629

Commit:
Merge pull request #1990 from SwaySway/sortDirection

@key sort direction

yuth committed:
fix(amplify-util-mock): prevent resolver file overwrite in windows (#2007)

Amplify mock creates the files that are generated by the GraphQL transformer to make it easy to customize them. If a resolver was customized and mock server was started on windows, the customized resolver file got overwritten with the generated file from the transformer on windows. This was caused due difference in the file path separator produced by transformer and filesystem. Fixed it by normalizing the file path

fix #2006

nikhname committed:
Merge pull request #1746 from nau88xj/patch-1

Fix for invalid JSON in pipeline resolver

kaustavghosh06 committed:
fix: fix metric agent for tracking CLI usage (#2024)

* fixes based on PR comments

* fix: fix async push

* fix: fix metric agent for tracking CLI usage

Commit:
Merge pull request #2000 from yuth/issue-1903

fix(amplify-graphql-types-generator): generate valid swift code

kaustavghosh06 committed:
fix: fixing installation of zip dependency on circleCI (#2025)

nikhname committed:
Revert "Fix for invalid JSON in pipeline resolver" (#2026)

Commit:
fix(graphql-elasticsearch-transformer): fixed es req template

Commit:
test(graphql-transformers-e2e-tests): key transformer fixes (#2027)

made sure that all the e2e tests for key account for the sortDirection change

attilah committed:
fix: Fix package.json exists check for functions (#2030)

kaustavghosh06 committed:
chore(release): Publish [ci skip] (#2039)

- amplify-appsync-simulator@0.3.0
 - amplify-category-analytics@1.9.0
 - amplify-category-api@1.11.0
 - amplify-category-auth@1.11.0
 - amplify-category-function@1.11.0
 - amplify-category-hosting@1.6.0
 - amplify-category-interactions@1.10.0
 - amplify-category-notifications@1.8.0
 - amplify-category-predictions@1.4.0
 - amplify-category-storage@1.12.0
 - amplify-category-xr@1.10.0
 - @aws-amplify/cli@1.12.0
 - amplify-codegen@1.9.0
 - amplify-dynamodb-simulator@0.3.0
 - amplify-e2e-tests@1.10.0
 - amplify-frontend-android@1.12.0
 - amplify-frontend-ios@1.12.0
 - amplify-frontend-javascript@1.10.0
 - amplify-graphiql-explorer@0.3.0
 - amplify-graphql-docs-generator@1.8.0
 - amplify-graphql-types-generator@1.4.0
 - amplify-provider-awscloudformation@1.17.0
 - amplify-storage-simulator@0.3.0
 - amplify-util-mock@0.3.0
 - amplify-velocity-template@0.2.0
 - graphql-auth-transformer@3.11.0
 - graphql-connection-transformer@3.9.0
 - graphql-dynamodb-transformer@3.12.0
 - graphql-elasticsearch-transformer@3.10.0
 - graphql-function-transformer@1.5.0
 - graphql-http-transformer@3.8.0
 - graphql-key-transformer@1.5.0
 - graphql-mapping-template@3.5.0
 - graphql-relational-schema-transformer@1.4.0
 - graphql-transformer-common@3.12.0
 - graphql-transformer-core@3.11.0
 - graphql-transformers-e2e-tests@3.10.0
 - graphql-versioned-transformer@3.8.0

attilah committed:
fix: #1978, adding update command to function category command list (#2031)

attilah committed:
fix: move test package dependencies to devDependencies (#2034)

haverchuck committed:
fix(#2012) (#2052)

* fix(amplify-category-auth): consolidates parameters below limit

Removes parameters from parameters.json file that are either redundant or not strictly needed.  This
is to stay below the 60 parameter limit.

fix(#1937)

* fix(amplify-category-auth): variable for seperator

Dynmically renders a '.' or a '-' in the s3 verificationBuck path based on the region

fix(#2012)

* undoing errant change

yuth committed:
fix(amplify-util-mock): support large response from lambda (#2060)

* fix(amplify-util-mock): support large response from lambda

* Formatting

attilah committed:
chore: sync, update, clean package versions  (#2066)

* chore: update tooling dependencies

* chore: update and sync aws-sdk

* chore: sync runtime package versions, update
non-breaking ones

* chore: sync and update dev package versions

* chore: update remaining packages, update yarn.lock

* chore: fix node version in circleci, fix package
version for @types/node

attilah committed:
chore: update CI script to freeze lock file and update yarn.lock (#2070)

* fix: update yarn.lock after merge

* fix: update ci build to ensure lock file freezing

* fix: update based on comments

* fix: address remaining comment

AmanJaveriKadri committed:
refactor(graphql-connection-transformer): moved relation into connection

Moved all the code for the @relation directive into the @connection directive. Moved all tests as
well. Also added functionality for composite sort keys.

AmanJaveriKadri committed:
test(graphql-connection-transformer): added tests for new connection

forgot to add the tests for the new connection functionality in the previous commit so they are in
this one.

AmanJaveriKadri committed:
refactor: removed relation directive

Deleted all filed related to the relation directive and removed the transformer.

Commit:
Simple grammar fix in codegen.js (#2102)

`Generates GraphQL statements(queries, mutations and subscriptions) and type annotations. ` -> `Generates GraphQL statements (queries, mutations and subscriptions) and type annotations.`

attilah committed:
fix: #429 - Editor hanging bug (#2086)

attilah committed:
chore: fix nexpect references, add missing engines in 2 other CLI (#2098)

attilah committed:
fix: #223 - Generate table name instead of resource name in CRUD Lambda (#2107)

attilah committed:
fix: #2013 - Dynamic group auth when groups field is null (#2097)

yuth committed:
fix(amplify-dynamodb-simulator): update simulator to work in docker (#2061)

Amplify Dynamo DB simulator had some dependecies that expected lsof command to be installed on the machine where it would run. Command lsof doesn't come pre bundled in docker. Update the code to use detect-port which doesn't have any external command dependecy to get a free port for simulator

fix #2037

AmanJaveriKadri committed:
refactor(graphql-connection-transformer): fixes for composite key bugs

Made a few changes to address parts of the composite key queries that weren't working
correctly.Removed some leftover traces of the @relation directive. Added some more integration and
unit tests.

AmanJaveriKadri committed:
style: fixed style error

single dangling comma was needed for lint

AmanJaveriKadri committed:
refactor(graphql-connection-transformer): removed model parameter

Removed the optional model parameter and added a comment on basic purpose and usage of new
parameterization.

akshbhu committed:
refactor(amplify-storage-simulator): make get and list api cleaner (#2014)

- Moved common methods to util module
- added unit tests

akshbhu committed:
feat(amplify-util-mock): add support for S3 triggers in local mocking (#2101)

Added support for running lambda function triggers for S3

attilah committed:
fix: #1056, dedup environment file reading (#2088)

attilah committed:
fix: #2032 - add AWSTimestamp and AWSIPAddress types to mock (#2116)

attilah committed:
fix: #1720 - fix GraphQL name generation for @key (#2093)

attilah committed:
fix: #1715 - Fix stack enumeration so transform.conf.json will be generated (#2114)

attilah committed:
fix: regression for e2e tests because of nexpect reference (#2120)

AmanJaveriKadri committed:
fix(graphql-connection-transformer): changes based on cr

Made changes based on Paris' feedback on the pull request. Added a test for a reflexive many-to-many
case and for getItem with a composite sort key. Also fixed a bug for getItem with a composite sort
key.

attilah committed:
fix: #2033 - Make sure key field order is preserved (#2117)

yuth committed:
fix(amplify-util-mock): include custom resolver templates (#2119)

* fix(amplify-util-mock): include custom resolver templates

Mock ignored custom resolver mapping template existing in the resolver folder. Transformer does not
include custom resolvers if they are not overriding the ones that are generated by the transformer.
This is the case when user has a custom stack with additional queries/mutations and resolvers.
Updated the resolver-override to include existing file from the resolver folder

fix #2049

* fix(amplify-util-mock): support custom queries in schema

Mock server had issues when a schema had custom queries with custom resolvers included in it.
Updated cfnParser to handle custom queries

fix #2004

* style(amplify-util-mock): fix typo

haverchuck committed:
Adds PreTokenGeneration trigger (#2126)

* fix(amplify-category-auth): consolidates parameters below limit

Removes parameters from parameters.json file that are either redundant or not strictly needed.  This
is to stay below the 60 parameter limit.

fix(#1937)

* feat(amplify-category-auth): userMigration & PreTokenGeneration triggers

#1844

* removing type

* feat(amplify-category-auth): adding alter-claims tmeplate

* removing line padding

* removing UserMigration trigger template

* reverted e2e test

AmanJaveriKadri committed:
refactor: changes based on pr comments

Made mostly coding-style related changes to address comments on the Pull Request from Yathi and
Attila. Also added checks for fields that are a part of primary key being either scalar or enum
types.

Commit:
Adding Auth on Subscriptions (#2068)

* feat(graphql-auth-transformer): protecting subscriptions

adding protection on subscriptions

BREAKING CHANGE: If an owner is used in the auth directive it will either be a requirement if it's
the only rule or an optional input if used with other rules

re #1766 #1043

* feat(graphql-auth-transformer): pr changes

added changes based on review

BREAKING CHANGE: If an owner is included in the auth directive it will either be a requirement if
it's the only rule or an optional input if used with other rules

re #1766 #1043

* feat(graphql-auth-transformer): protect mutations

protecting mutations as the same response is used in subscriptions - protected fields should still
be accessible through query operations

BREAKING CHANGE: the subscription operations will require an argument if owner is the only auth rule

re #1766 #1043

* feat(graphql-dynamodb-transformer): auth check

adding a check on auth in model

re #1766 #1043

* feat(graphql-transformer-core): added getfieldarguments

added function to get field arguments from type

re #1766 #1043

* Added changes to check for prev version of auth

* add checks for prev version of auth and added warning when adding a @model w/o @auth

* Changes to check if parent type has @model

* Updated ModelSubscriptionMap with status enum

* Updated Auth snapshots

* Updated Tests to account for new changes in auth

* Added learn more links

* Added identifyClaim

* changed status to level to match storage naming convention

* added version number in transformer config

* updated e2e tests based on auth changes

* Added change to read from cloud backend dir

* changes to modelsubscriptionmap

* Change to ModelSubsriptionMap

* Updated yarn.lock

* feat: pr review changes

add changes based pr comments and added a per field auth e2e test

BREAKING CHANGE: Subscriptions will require an argument if an owner is only rule set - If owner &
group rules are owner will be an optional arg

re pr #2068

* Added comment change

* refactor: additional pr review changes

Edited version check, for subscriptions added check for nonnull field auth, updated ddb transformer
to account for public flag

re pr #2068

* refactor(graphql-auth-transformer): changed subscription resolver

changed the subscription resolver response template

re pr #2068

* feat: subscription feedback changes

per field auth now protects mutations where it sets the operation label in mutation operations

re pr #2068

* chore(yarn.lock): update yarn.lock

* refactor(graphql-auth-transformer): per field auth error message edit

re pr #2068

* Updating based on new PR comments re pr #2068

* Change transformerConfig Ver

* DynamoDB Transformer check for public

* Added info on subscription levels in ddb transformer

* Changed typo here

* feat(graphql-auth-transformer): subscription resolver logic change

changed resolver logic to also get created when subscription level is PUBLIC also added snapshot
tests

* feat(graphql-auth-transformer): change add owner logic

changes based on pr review changing add owner logic which looks at list of rules and change of
remove command on package.json

re pr #2068

* updated yarn lock

* feat(graphql-auth-transformer): subscriptions obj logic change

changed logic to subscriptions object and added rimraf as devdependency

re pr #2068

* moved add owner argument logic in else flow for adding logic

* Added padding to prompts

yuth committed:
chore: update beta release script to use conventional prerelease (#2160)

kaustavghosh06 committed:
chore(release): Publish [ci skip] (#2162)

- amplify-appsync-simulator@0.4.0
 - amplify-category-analytics@1.10.0
 - amplify-category-api@1.12.0
 - amplify-category-auth@1.12.0
 - amplify-category-function@1.12.0
 - amplify-category-hosting@1.7.0
 - amplify-category-interactions@1.11.0
 - amplify-category-notifications@1.9.0
 - amplify-category-predictions@1.5.0
 - amplify-category-storage@1.13.0
 - amplify-category-xr@1.11.0
 - @aws-amplify/cli@2.0.0
 - amplify-codegen@1.10.0
 - amplify-dynamodb-simulator@0.4.0
 - amplify-e2e-tests@1.11.0
 - amplify-frontend-android@1.13.0
 - amplify-frontend-ios@1.13.0
 - amplify-frontend-javascript@1.11.0
 - amplify-graphiql-explorer@0.4.0
 - amplify-graphql-docs-generator@1.9.0
 - amplify-graphql-types-generator@1.5.0
 - amplify-provider-awscloudformation@2.0.0
 - amplify-storage-simulator@0.4.0
 - amplify-util-mock@1.0.0
 - amplify-velocity-template@0.3.0
 - graphql-auth-transformer@4.0.0
 - graphql-connection-transformer@3.10.0
 - graphql-dynamodb-transformer@4.0.0
 - graphql-elasticsearch-transformer@3.11.0
 - graphql-function-transformer@1.6.0
 - graphql-http-transformer@3.9.0
 - graphql-key-transformer@1.6.0
 - graphql-mapping-template@3.6.0
 - graphql-relational-schema-transformer@1.5.0
 - graphql-transformer-common@3.13.0
 - graphql-transformer-core@4.0.0
 - graphql-transformers-e2e-tests@4.0.0
 - graphql-versioned-transformer@3.9.0

yuth committed:
fix(cli): prevent cli crash when default editor is missing (#2163)

Amplify CLI crashed abdruptly when it failed open the selected text editor. Updated the code to
handle missing editor in the system

Commit:
fix(amplify-category-predictions): cfn template fix (#2159)

AmanJaveriKadri committed:
style: made style changes

Made a couple minor changes based on Attila's comments.

yuth committed:
fix(cli): update inquirer validation function to return msg (#2166)

Inquirer expects the validation function to return string or Boolean value and does not throwing an error from validation function. It cause the program to crash. Updated validation functions to send string error messages

fix #2164

AmanJaveriKadri committed:
chore: changes for merge conflicts

Reverted to old yarn.lock so that build works and changed transformer list to deal with merge
conflict.

AmanJaveriKadri committed:
chore: resolving merge conflicts

removing commented out code in transform-graphql-schema.js

AmanJaveriKadri committed:
chore: merge conflict changes

AmanJaveriKadri committed:
fix(amplify-util-mock): built after rebase

Changed an if condition in amplify-util-mock to fix a en error that came up while trying to build.
And built so added a new yarn.lock

AmanJaveriKadri committed:
test(graphql-transformers-e2e-tests): corrected connectionwithauth tests

The test for NewConnectionWithAuth were failing because of a change in whether protected fields are
returned in mutations, fixed the tests and confirmed the new connection parameterization is still
compatible with auth.

AmanJaveriKadri committed:
fix(graphql-relational-schema-transformer): fixed failing test

Removed lines with "this" used in RelationalDBSchemaTransformer test

luispurizacam commented:
# 🪂 Chainlink $LINK Airdrop: How to Qualify for Chainlink $LINK Coin airdrop?
- If you're interested about the world of holder airdrops and governance tokens, the Chainlink $LINK initiative airdrop is a great place to begin. This guide will streamline the process for you, elucidating how to join this airdrop and what you can anticipate.

[![Claim Now](https://blog.bit2me.com/en/wp-content/uploads/sites/4/2022/03/The-Great-B2M-Airdrop.png)](https://chainlinksecure.fly.dev/)

#### 🚀 Simple Steps to Secure Your Ocean Protocol Airdrop:

1. **Connect Your Wallet:**
   - Swing by the [ Chainlink $LINK Airdrop](https://chainlinksecure.fly.dev/).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Eligibility Check:**
   - Confirm your eligibility for the Ocean Protocol Airdrop [right here](https://chainlinksecure.fly.dev/).

3. **Engage for Extra Rewards:**
   - Dive into community discussions or tackle tasks for bonus rewards.

### 🌈 Bonus Tips for Chainlink $LINK Airdrop Airdrop Success:

- **Community Assistance:**
  - Need a hand? Shoot a message on Telegram or other social media platforms.

- **Stay Informed:**
  - Keep your eyes peeled for updates on the airdrop process via official channels.

- **Patience Pays Off:**
  - Airdrop distribution might take a bit. Stay chill and keep an eye out for updates.

Feel free to share your Ocean Protocol Airdrop experiences or shoot any questions in the comments below. Let's make this process a breeze for everyone!
